src/compass/s1_cslc_qa.py

```diff
@@ -0,0 +1,144 @@
+
+            # create dict for current polarization
+            self.stats_dict[pol] = {}
+            pol_dict = self.stats_dict[pol]
+
+            # write stats to HDF5
+            for real_imag, cstat_member in zip(['real', 'imag'],
+                                               [stat_obj.real, stat_obj.imag]):
+                # create dict to store real/imaginary stat items
+                pol_dict[real_imag] = {}
+
+                vals = [cstat_member.mean, cstat_member.min,
+                        cstat_member.max, cstat_member.sample_stddev]
+                # write stats to HDF5
+                for val_name, val in zip(self.stat_names, vals):
+                    pol_dict[real_imag][val_name] = val
+
+
+    def compute_correction_stats(self, cslc_h5py_root):
+        '''
+        Compute correction stats
+
+        Parameters
+        ----------
+        cslc_h5py_root: h5py.File
+            Root of CSLC HDF5
+        '''
+        # compute and save corrections stats
+        corrections = ['bistatic_delay', 'geometry_steering_doppler',
+                       'azimuth_fm_rate_mismatch']
+
+        # compute stats and write to hdf5
+        self.stats_dict['corrections'] = {}
+        corrections_dict = self.stats_dict['corrections']
+        for correction in corrections:
+            # create dict for current correction
+            corrections_dict[correction] = {}
+            correction_dict = corrections_dict[correction]
+
+            # get dataset and compute stats according to dtype
+            src_path = f'{ROOT_PATH}/corrections/{correction}'
+            ds = cslc_h5py_root[src_path]
+            stat_obj = isce3.math.StatsFloat32(ds[()].astype(np.float32))
+
+            # write stats to HDF5
+            vals = [stat_obj.mean, stat_obj.min, stat_obj.max]
+            for val_name, val in zip(self.stat_names, vals):
+                correction_dict[val_name] = val
+
+
+    def raster_pixel_classification(self):
+        '''
+        Place holder for populating classification of geocoded pixel types
+        '''
+        self.classification_count_dict['topo'] = {}
+        topo_dict = self.classification_count_dict['topo']
+        topo_dict['layover_percent'] = 0
+        topo_dict['shadow_percent'] = 0
+        topo_dict['combined_percent'] = 0
+        self.classification_count_dict['percent_land_pixels'] = 0
+        self.classification_count_dict['percent_valid_pixels'] = 0
+
+
+    def populate_rfi_dict(self):
+        '''
+        Place holder for populating SAFE RFI information
+        '''
+        self.rfi_dict['is_rfi_info_available'] = True
+        # Follow key/values only assigned if RFI info is avaiable
+        self.rfi_dict['rfi_mitigation_performed'] = True
+        self.rfi_dict['rfi_mitigation_domain'] = True
+        self.rfi_dict['rfi_burst_report'] = True
+
+
+    def populate_orbit_dict(self):
+        '''
+        Place holder for populating QA orbit information
+        '''
+        pass
```

Comment:
@vbrancat @seongsujeong @scottstanie I know there are different types of orbits but I am ignorant of any specifics. Can any of you help me populate this piece of QA? Thanks

Reply:
[here describes the names](https://sentinel.esa.int/web/sentinel/-/copernicus-sentinel-1a-and-1b-precise-orbital-products-currently-available-for-the-entire-mission/1.2?redirect=%2Fweb%2Fsentinel%2Fmissions%2Fsentinel-1)

[here describes their accuracies](https://sentinel.esa.int/web/sentinel/technical-guides/sentinel-1-sar/pod/quality-control/orbit-comparisons)

- AUX_PREORB is the prediction one that comes with the product
- AUX_RESORB: restituted orbit files 
 > generated within 180 min from sensing time and distributed daily, with accuracy requirement of 10 cm in 2D RMS, but typically below 5 cm. A rolling policy of 1 month is applied, in order to keep only the latest files.
- AUX_POEORB: precise orbit files 
> generated with a timeliness of 20 days from sensing time and distributed daily, with accuracy requirement of 5 cm in 3D RMS, but typically below 1 cm. Once available, these files supersede the AUX_RESORB products, as they contain the same data with better accuracy. These products are available from the beginning of the mission.

Reply:
I have alreaday convinced myself it's fine to have the orbit quality in the "QA" group to please ignore that last comment :)